olmo/train.py

```diff
@@ -145,22 +146,30 @@ class Trainer:
 
     def __post_init__(self):
         if self.cfg.fused_loss:
+            import flash_attn
             from flash_attn.ops.triton.cross_entropy import (  # type: ignore
                 cross_entropy_loss,
             )
 
             def fused_loss_fn(
                 logits, labels, ignore_index: int = -100, reduction: str = "mean", compute_z_loss: bool = False
             ):
+                ignore_index_kwarg = {}
+                if version.parse(flash_attn.__version__) >= version.parse("2.5.8"):
+                    # The parameter name was changed in v2.5.8 with commit https://github.com/Dao-AILab/flash-attention/commit/ec6d22143b5d375e253b2ebfc563b26a43f43684
+                    ignore_index_kwarg = {"ignore_index": ignore_index}
+                else:
+                    ignore_index_kwarg = {"ignored_index": ignore_index}
+
```

Comment:
I think it would be more efficient to parse the version outside of this function, above, so we don't do it on every call.

Reply:
[74d3efe](https://github.com/allenai/OLMo/pull/580/commits/74d3efe8be29b61f42411ddbf0ccce3dd76a0265)